//
//           ┌┬┐┬ ┬┌─┐┬ ┬┌─┐┌─┐┌┐┌
//            │ └┬┘├─┘├─┤│ ││ ││││
//            ┴  ┴ ┴  ┴ ┴└─┘└─┘┘└┘
//
// This file is part of the typhoon Project.
// Copyright (C) 2021 stanley0207@163.com
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

syntax = "proto3";

import "google/protobuf/descriptor.proto";

package tpn.protocol;

option optimize_for = SPEED;

option cc_generic_services = false;

// 扩展服务选项
extend google.protobuf.ServiceOptions {
  tpn.protocol.TPNServiceOptions service_options = 90000;
}

// 扩展方法选项
extend google.protobuf.MethodOptions {
  tpn.protocol.TPNMethodOptions method_options = 90000;
}

// 服务选项
message TPNServiceOptions {
  string descriptor_name = 1;  // 用于生成 service_hash值
}

// 方法选项
message TPNMethodOptions {
  uint32 id = 1;  // 方法id
}
